janus_server/src/datastore.rs

```diff
@@ -1305,10 +1303,109 @@ impl<C: Clock> Transaction<'_, C> {
             )
             .await?
             .into_iter()
-            .map(|row| report_aggregation_from_row(vdaf, role, task_id, aggregation_job_id, row))
+            .map(|row| {
+                let report_id_bytes: [u8; ReportId::LEN] = row
+                    .get::<_, Vec<u8>>("report_id")
+                    .try_into()
+                    .map_err(|err| {
+                        Error::DbState(format!("couldn't convert report_id value: {err:?}"))
+                    })?;
+                let report_id = ReportId::from(report_id_bytes);
+                Self::report_aggregation_from_row(
+                    vdaf,
+                    role,
+                    task_id,
+                    aggregation_job_id,
+                    &report_id,
+                    &row,
+                )
+            })
             .collect()
     }
 
+    fn report_aggregation_from_row<const L: usize, A: vdaf::Aggregator<L>>(
+        vdaf: &A,
+        role: &Role,
+        task_id: &TaskId,
+        aggregation_job_id: &AggregationJobId,
+        report_id: &ReportId,
```

Comment:
Hmm, all the other `_from_row` methods use references, except for `aggregation_job_from_row` (and one argument to `task_from_rows` that was missed somehow). Updated.

FTR, I care a lot less about internal (private) functions than I do about exported-from-module (public) functions. Even for private functions it's still probably a good idea to follow along to avoid unnecessary copies, but ultimately the practical needs of the module would overweigh any particular calling convention (so to speak). For these methods in particular many of the arguments don't matter much -- other than `row`, most of them end up getting copied into the eventual structure being created anyway. Passing a reference for these arguments at most saves a copy if we short-circuit on some error before we get to creating the structure.